File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        int n = nums.length;
        Set<Integer>st = new HashSet<>();
        long sum = 0, maxi_sum = 0;

        int i = 0; 
        for(int j = 0; j < n; j++){

            // Check the element present in set or not??
            while(st.contains(nums[j])){
                st.remove(nums[i]);
                sum -= nums[i];
                i++;
            }
            // element is not present in set. Then, add into set
            st.add(nums[j]);
            sum += nums[j];

            // Now, my current window size is "k-sized". Calculate the maximum & shrink window
            if(j-i+1 == k){
                maxi_sum = Math.max(maxi_sum, sum);
                st.remove(nums[i]);
                sum -= nums[i];
                i++;
            }
        }
        return maxi_sum;
    }
}

File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.cpp
class Solution {
public:
    long long maximumSubarraySum(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_set<int>st;
        long long sum = 0, maxi_sum = 0;

        int i = 0;
        for(int j = 0; j < n; j++){

            // Check the element present in the set or not??
            while(st.find(nums[j]) != st.end()){
                st.erase(nums[i]);
                sum -= nums[i];
                i++;
            }
            // Or else, add into set
            st.insert(nums[j]);
            sum += nums[j];

            // if my window size is "k-sized". Then, calculate sum & keep maximum shrink window
            if(j-i+1 == k){
                maxi_sum = max(maxi_sum, sum);
                st.erase(nums[i]);
                sum -= nums[i];
                i++;
            }
        }
        return maxi_sum;
    }
};